### PR TITLE
visidata: fix #202165, add shell completions

### LINKED ISSUE #202165: visidata, pudb and trash-cli miss shell completions
### Describe the bug
<https://github.com/NixOS/nixpkgs/blob/nixos-22.05/pkgs/tools/misc/trash-cli/default.nix> and <https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/python-modules/pudb/default.nix> miss shell completion.

### Expected behavior
See <https://github.com/archlinux/svntogit-community/blob/packages/trash-cli/trunk/PKGBUILD#L31-L34> and <https://docs.iterative.ai/shtab/use/>

### PR DESCRIPTION
**Pull request by Freed-Wu**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/visidata/default.nix [hunk 1]
```diff
@@ -130,6 +130,10 @@ buildPythonApplication rec {
     bash dev/test.sh
     runHook postCheck
   '';
+  postInstall = ''
+    python dev/zsh-completion.py
+    install -Dm644 _${pname} -t $out/share/zsh/site-functions
```
Comment (superherointj) [inline pkgs/applications/misc/visidata/default.nix:135]: No bash completion?

Comment (superherointj) [inline pkgs/applications/misc/visidata/default.nix:135]:
```suggestion
    install -Dm644 _visidata -t $out/share/zsh/site-functions
```

Comment (Freed-Wu) [reply pkgs/applications/misc/visidata/default.nix:135]: Yes <https://github.com/saulpw/visidata/commit/6ffda2adda730702e0bfe9f4fcb2d375e8d920e1>

Comment (Freed-Wu) [reply pkgs/applications/misc/visidata/default.nix:135]: Why? It has no difference.

Comment (superherointj) [reply pkgs/applications/misc/visidata/default.nix:135]: > Why? It has no difference.

`_visidata` is more clear. Avoids unnecessary abstraction. It's static. etc.

**Review by superherointj — APPROVED** [review]

**Comment by Freed-Wu** [comment]
Maybe we can use `installShellCompletion`: <https://github.com/caarlos0/nixpkgs/blob/28c08a222e13bf9533e3f92fb54af08c1658a509/pkgs/tools/package-management/nfpm/default.nix#L31>

**Pull request merged**